### PR TITLE
Migrate slack-send-file command in SlackV3 Integration to new APIs

### PR DESCRIPTION
**Pull request by kamalq97**
## Related Issues
fixes: [link to the issue](https://jira-dc.paloaltonetworks.com/browse/CIAC-11506)

## Description
- Updated the **slack-send-file** command to be compatible with all apps - including newly created apps from May 8, 2024.

### CODE REVIEW

File: Packs/Slack/Integrations/SlackV3/SlackV3.py [hunk 1]
```diff
@@ -55,6 +56,17 @@
 PROFILING_DUMP_ROWS_LIMIT = 20
 MAX_SAMPLES = 10
 
+FileUploadParams = namedtuple(
```
Comment (kamalq97) [inline Packs/Slack/Integrations/SlackV3/SlackV3.py:59]: Better to update declaration of `namedtuple` to be a class with typing hints (according to @shmuel44)

Comment (kamalq97) [reply Packs/Slack/Integrations/SlackV3/SlackV3.py]: Fixed in commit a16dac6b5d1baf5bffc7d6e6544be2d1ae7dad4b

File: Packs/Slack/Integrations/SlackV3/SlackV3.py [hunk 2]
```diff
@@ -458,17 +468,22 @@ def set_name_and_icon(body: dict, method: str):
...
     """
     Sends a request to slack API while handling rate limit errors.
 
     Args:
         client: The slack client.
         method: The method to use.
         http_verb: The HTTP method to use.
-        file_: A file path to send.
         body: The request body.
+        file_upload_params: An instance of FileUploadParams.
```
Comment (shmuel44) [inline Packs/Slack/Integrations/SlackV3/SlackV3.py:486]: Please add to the docstring that this argument is related only to the file upload command.

File: Packs/Slack/Integrations/SlackV3/SlackV3.py [hunk 3]
```diff
@@ -55,6 +56,15 @@
 PROFILING_DUMP_ROWS_LIMIT = 20
 MAX_SAMPLES = 10
 
+
+class FileUploadParams(NamedTuple):
```
Comment (shmuel44) [inline Packs/Slack/Integrations/SlackV3/SlackV3.py:60]: Let's use `TypedDict` instead of `NamedTuple`.

File: Packs/Slack/Integrations/SlackV3/SlackV3.py [hunk 4]
```diff
@@ -528,8 +548,8 @@ async def send_slack_request_async(client: AsyncWebClient, method: str, http_ver
         try:
             demisto.debug(f'Sending slack {method} (async). Body is: {str(body)}')
             if http_verb == 'POST':
-                if file_:
-                    response = await client.api_call(method, files={"file": file_}, data=body)  # type: ignore
+                if file_upload_params:
+                    response = await client.files_upload_v2(**file_upload_params._asdict())  # type: ignore
                 else:
                     response = await client.api_call(method, json=body)  # type: ignore
```
Comment (shmuel44) [inline Packs/Slack/Integrations/SlackV3/SlackV3.py:554]:
```suggestion
                if file_upload_params:
                    response = await client.files_upload_v2(**file_upload_params._asdict())
                else:
                    response = await client.api_call(method, json=body)
```

File: Packs/Slack/Integrations/SlackV3/SlackV3.py [hunk 5]
```diff
@@ -2005,15 +2025,11 @@ def slack_send_file(_channel: str | None = None, _channel_id: str = '', _entry_i
...
-        else:
-            demisto.results('Could not send the file to Slack.')
-    except SlackApiError as e:
-        if 'method_deprecated' in str(e):
-            raise DemistoException('Command slack-send-file isn\'t available for newly created apps (from May 8, 2024).')
+    response = slack_send_request(to, channel, group, thread_id=thread_id, file_dict=file_dict, channel_id=channel_id)
+    if response and response.get('ok'):
+        demisto.results('File sent to Slack successfully.')
+    else:
+        demisto.results('Could not send the file to Slack.')
```
Comment (shmuel44) [inline Packs/Slack/Integrations/SlackV3/SlackV3.py:2032]:
```suggestion
        return_results(CommandResults('File sent to Slack successfully.'))
    else:
        return_results(CommandResults('Could not send the file to Slack.'))
```

File: Packs/Slack/Integrations/SlackV3/SlackV3.py [hunk 6]
```diff
@@ -2167,19 +2183,16 @@ def send_file_to_destinations(destinations: list, file_dict: dict, thread_id: st
...
-        body['initial_comment'] = file_dict['comment']
 
     for destination in destinations:
-        body['channels'] = destination
-        if thread_id:
-            body['thread_ts'] = thread_id
-
-        response = send_slack_request_sync(CLIENT, 'files.upload', file_=file_dict['path'], body=body)
+        file_upload_params = FileUploadParams(
+            filename=file_dict['name'],  # Mandatory
```
Comment (shmuel44) [inline Packs/Slack/Integrations/SlackV3/SlackV3.py:2189]:
```suggestion
            filename=file_dict['name'],
```

File: Packs/Slack/Integrations/SlackV3/SlackV3.py [hunk 7]
```diff
@@ -2167,19 +2183,16 @@ def send_file_to_destinations(destinations: list, file_dict: dict, thread_id: st
...
+            file=file_dict['path'],
+            initial_comment=file_dict.get('comment'),  # Optional
```
Comment (shmuel44) [inline Packs/Slack/Integrations/SlackV3/SlackV3.py:2191]:
```suggestion
            initial_comment=file_dict.get('comment'),
```

**Review by shmuel44 — CHANGES_REQUESTED** [review]
Nice!

File: Packs/Slack/ReleaseNotes/3_4_20.md [hunk 8]
```diff
@@ -0,0 +1,6 @@
+
+#### Integrations
+
+##### Slack v3
+
+- Updated the **slack-send-file** command to be compatible with all apps - including newly created apps from May 8, 2024.
```
Comment (ShirleyDenkberg) [inline Packs/Slack/ReleaseNotes/3_4_20.md:6]:
```suggestion
Updated the ***slack-send-file*** command to be compatible with all apps - including newly created apps from May 8, 2024.
```

**Comment by ShirleyDenkberg** [comment]
@JasBeilin @dorschw @rshunim @amshamah419 @shmuel44 Doc review completed.

File: Packs/Slack/ReleaseNotes/3_4_20.md (hunk 8 above)
Comment (kamalq97) [reply Packs/Slack/ReleaseNotes/3_4_20.md]: Thank you for the feedback, I have now updated the Release Notes based on your suggestion

File: Packs/Slack/Integrations/SlackV3/README.md [hunk 9]
```diff
@@ -283,16 +283,28 @@ There is no context output for this command.
...
 
+#### Permissions
+
+Scopes required:
+
+| **Token Type**    | **Scope**                     |
+| --- | --- | 
+| Granular bot      | files:write                   |
+| User              | files:write files:write:user  |
+| Legacy bot        | bot                           |
```
[the remark this answers is not in the document]
Comment (kamalq97) [reply Packs/Slack/Integrations/SlackV3/README.md:298]: Added a link in commit af64ee781d6cc83d3a13f2639f44c91cbf6fa8aa
Please let me know if additional changes are required to the `README.md`

File: Packs/Slack/Integrations/SlackV3/SlackV3.py [hunk 10]
```diff
@@ -528,12 +586,12 @@ async def send_slack_request_async(client: AsyncWebClient, method: str, http_ver
         try:
             demisto.debug(f'Sending slack {method} (async). Body is: {str(body)}')
             if http_verb == 'POST':
-                if file_:
-                    response = await client.api_call(method, files={"file": file_}, data=body)  # type: ignore
+                if file_upload_params:
```
[the remark this answers is not in the document]
Comment (kamalq97) [reply Packs/Slack/Integrations/SlackV3/SlackV3.py:590]: Added an explanatory comment in commit af64ee781d6cc83d3a13f2639f44c91cbf6fa8aa

File: Packs/Slack/Integrations/SlackV3/README.md [hunk 11]
```diff
@@ -283,16 +283,30 @@ There is no context output for this command.
...
+
+#### Limitations
+
+There are no known limitations for this command.
+
 #### Input
 
 | **Argument Name** | **Description**                                                                                     | **Required** |
-|-------------------|-----------------------------------------------------------------------------------------------------|--------------|
+| --- | --- | --- |
```
Comment (dorschw) [inline Packs/Slack/Integrations/SlackV3/README.md:309]: nice

File: Packs/Slack/Integrations/SlackV3/README.md [hunk 12]
```diff
@@ -283,16 +283,30 @@ There is no context output for this command.
 Sends a file to a user, channel, or group. If not specified, the file is sent to the mirrored investigation channel (if
 the channel exists).
 
-**Note:** Command slack-send-file isn't available for newly created apps (from May 8, 2024) and will get an error in case of use, due to changes from the Slack API side.
```
Comment (dorschw) [inline Packs/Slack/Integrations/SlackV3/README.md:286]: good catch

File: Packs/Slack/Integrations/SlackV3/SlackV3.py [hunk 13]
```diff
@@ -28,6 +29,38 @@
     'Critical': 4
 }
 
+RESPONSE_ERROR_EXPLANATIONS = {
```
Comment (dorschw) [inline Packs/Slack/Integrations/SlackV3/SlackV3.py:32]: since this dictionary is only used in one place (and is unclear when just reading the file), either move it to the same function where it's used (preferred imo), or just above the function

File: Packs/Slack/Integrations/SlackV3/SlackV3.py [hunk 14]
```diff
@@ -55,6 +88,15 @@
 PROFILING_DUMP_ROWS_LIMIT = 20
 MAX_SAMPLES = 10
 
+
+class FileUploadParams(TypedDict):
+    filename: str
+    file: str
```
Comment (dorschw) [inline Packs/Slack/Integrations/SlackV3/SlackV3.py:94]: what does `file` represent? the path? the content?

File: Packs/Slack/Integrations/SlackV3/SlackV3.py [hunk 15]
```diff
@@ -458,17 +500,22 @@ def set_name_and_icon(body: dict, method: str):
             body['icon_url'] = BOT_ICON_URL
 
 
-def send_slack_request_sync(client: slack_sdk.WebClient, method: str, http_verb: str = 'POST', file_: str = '',
-                            body: dict = None) -> SlackResponse:
+def send_slack_request_sync(
+    client: slack_sdk.WebClient,
+    method: str = '',
```
Comment (dorschw) [inline Packs/Slack/Integrations/SlackV3/SlackV3.py:505]: why default `''`?

File: Packs/Slack/Integrations/SlackV3/SlackV3.py [hunk 16]
```diff
@@ -482,8 +529,9 @@ def send_slack_request_sync(client: slack_sdk.WebClient, method: str, http_verb:
         try:
             demisto.debug(f'Sending slack {method} (sync). Body is: {str(body)}')
             if http_verb == 'POST':
-                if file_:
-                    response = client.api_call(method, files={"file": file_}, data=body)
+                if file_upload_params:
+                    # Use three-stage file_upload_v2 'wrapper' method in SDK client class in case there are file_upload_params
```
Comment (dorschw) [inline Packs/Slack/Integrations/SlackV3/SlackV3.py:533]:
```suggestion
                    # When file_upload_params is provided, performs a three-stage `file_upload_v2` wrapper method in Slack's SDK client
```
Worth adding a link to Slack's docs about this change

File: Packs/Slack/Integrations/SlackV3/SlackV3.py [hunk 17]
```diff
@@ -504,17 +552,22 @@ def send_slack_request_sync(client: slack_sdk.WebClient, method: str, http_verb:
     return response  # type: ignore
 
 
-async def send_slack_request_async(client: AsyncWebClient, method: str, http_verb: str = 'POST', file_: str = '',
-                                   body: dict = None) -> SlackResponse:
+async def send_slack_request_async(
+    client: AsyncWebClient,
+    method: str = '',
```
Comment (dorschw) [inline Packs/Slack/Integrations/SlackV3/SlackV3.py:557]: why default `''`?

File: Packs/Slack/Integrations/SlackV3/SlackV3.py [hunk 18]
```diff
@@ -504,17 +552,22 @@ def send_slack_request_sync(client: slack_sdk.WebClient, method: str, http_verb:
...
     """
     Sends an async request to slack API while handling rate limit errors.
 
     Args:
         client: The slack client.
         method: The method to use.
         http_verb: The HTTP method to use.
-        file_: A file path to send.
         body: The request body.
+        file_upload_params: An instance of FileUploadParams (for file upload APIs).
```
Comment (dorschw) [inline Packs/Slack/Integrations/SlackV3/SlackV3.py:570]:
```suggestion
        file_upload_params: An instance of FileUploadParams (for uploading using the file-upload API).
```
right?

File: Packs/Slack/Integrations/SlackV3/SlackV3.py [hunk 19]
```diff
@@ -528,12 +581,13 @@ async def send_slack_request_async(client: AsyncWebClient, method: str, http_ver
         try:
             demisto.debug(f'Sending slack {method} (async). Body is: {str(body)}')
             if http_verb == 'POST':
-                if file_:
-                    response = await client.api_call(method, files={"file": file_}, data=body)  # type: ignore
+                if file_upload_params:
+                    # Use three-stage file_upload_v2 'wrapper' method in SDK client class in case there are file_upload_params
```
Comment (dorschw) [inline Packs/Slack/Integrations/SlackV3/SlackV3.py:585]:
```suggestion
                    # When file_upload_params is provided, performs a three-stage `file_upload_v2` wrapper method in Slack's SDK client
```

Comment (dorschw) [inline Packs/Slack/Integrations/SlackV3/SlackV3.py:585]: I get a deja-vu, can we put this `if` in some method and call it twice, instead of duplicating it?

File: Packs/Slack/Integrations/SlackV3/SlackV3.py [hunk 20]
```diff
@@ -2005,15 +2059,22 @@ def slack_send_file(_channel: str | None = None, _channel_id: str = '', _entry_i
         'name': file_path['name'],
         'comment': comment
     }
+
+    error_message = 'Could not send the file to Slack.'
```
Comment (dorschw) [inline Packs/Slack/Integrations/SlackV3/SlackV3.py:2063]:
```suggestion
    error_message = 'Failed sending the file to Slack.'
```

Comment (dorschw) [inline Packs/Slack/Integrations/SlackV3/SlackV3.py:2063]: can we add some information about the file, like its name, to the error message? Will make it easier for users to understand what went wrong

File: Packs/Slack/Integrations/SlackV3/SlackV3.py [hunk 21]
```diff
@@ -2005,15 +2059,22 @@ def slack_send_file(_channel: str | None = None, _channel_id: str = '', _entry_i
...
         if response:
-            demisto.results('File sent to Slack successfully.')
+            return_results(CommandResults(readable_output='File sent to Slack successfully.'))
         else:
-            demisto.results('Could not send the file to Slack.')
-    except SlackApiError as e:
-        if 'method_deprecated' in str(e):
-            raise DemistoException('Command slack-send-file isn\'t available for newly created apps (from May 8, 2024).')
+            raise DemistoException(message=error_message)
+    except SlackApiError as api_error:
```
Comment (dorschw) [inline Packs/Slack/Integrations/SlackV3/SlackV3.py:2070]:
```suggestion
    except SlackApiError as e:
```

File: Packs/Slack/Integrations/SlackV3/SlackV3.py [hunk 22]
```diff
@@ -2005,15 +2059,22 @@ def slack_send_file(_channel: str | None = None, _channel_id: str = '', _entry_i
...
+        demisto.debug(f'{error_message} {api_error}')
+        error_code = api_error.response.get('error')
+        if error_code:
```
Comment (dorschw) [inline Packs/Slack/Integrations/SlackV3/SlackV3.py:2073]:
```suggestion
        if error_code := api_error.response.get('error'):
```

File: Packs/Slack/Integrations/SlackV3/SlackV3_test.py [hunk 23]
```diff
@@ -3291,26 +3289,53 @@ def test_send_message_to_destinations(mocker):
...
 
-def test_send_file_to_destinations(mocker):
+def test_send_file_to_destinations_request_args(mocker):
+    """
+    Given:
+        - Sending file to a channel
+    When:
+        - Calling the send_slack_request_sync function
+    Then:
+        - Assert the function is called once and the correct arguments are sent to the function
```
Comment (dorschw) [inline Packs/Slack/Integrations/SlackV3/SlackV3_test.py:3299]: let's discuss the tests here

File: Packs/Slack/Integrations/SlackV3/SlackV3_test.py [hunk 24]
```diff
@@ -3506,7 +3531,7 @@ def test_send_file_no_args_investigation(mocker):
 
     # Assert
     assert SlackV3.slack_send_request.call_count == 1
-    assert success_results[0] == 'File sent to Slack successfully.'
+    assert success_results[0]['HumanReadable'] == 'File sent to Slack successfully.'
```
Comment (dorschw) [inline Packs/Slack/Integrations/SlackV3/SlackV3_test.py:3534]: was it not CommandResults earlier, and now it is?

File: Packs/Slack/Integrations/SlackV3/SlackV3_test.py [hunk 25]
```diff
@@ -5232,29 +5257,23 @@ def test_get_war_room_url_with_xsiam_from_alert_war_room(self, mocker):
...
     """
     from SlackV3 import slack_send_file
     # Set
     mocker.patch.object(demisto, 'args', return_value={})
     mocker.patch.object(demisto, 'investigation', return_value={'id': '999'})
-    mocker.patch.object(demisto, 'getIntegrationContext', side_effect=get_integration_context)
-    mocker.patch.object(demisto, 'setIntegrationContext', side_effect=set_integration_context)
     mocker.patch('SlackV3.slack_send_request', side_effect=SlackApiError('The request to the Slack API failed. (url: https://slack.com/api/files.upload)',
-                                                                         {'ok': False, 'error': 'method_deprecated'}))
+                                                                         {'ok': False, 'error': 'service_unavailable'}))
```
Comment (dorschw) [inline Packs/Slack/Integrations/SlackV3/SlackV3_test.py:5274]: why is this changed?

File: Packs/Slack/pack_metadata.json [hunk 26]
```diff
@@ -2,7 +2,7 @@
     "name": "Slack",
     "description": "Interact with Slack API - collect logs, send messages and notifications to your Slack team.",
     "support": "xsoar",
-    "currentVersion": "3.4.19",
+    "currentVersion": "3.4.20",
```
Comment (dorschw) [inline Packs/Slack/pack_metadata.json:5]: let's bump a minor, this change is more significant than a revision.

**Review by dorschw — COMMENTED** [review]
Nice!

File: Packs/Slack/Integrations/SlackV3/SlackV3.py (hunk 14 above)
Comment (kamalq97) [reply Packs/Slack/Integrations/SlackV3/SlackV3.py:94]: * `file` is the file path
* `filename` is the name of the file

I agree that the naming is unclear, but I had to keep it consistent with the `slack_sdk.WebClient.file_upload_v2` kwargs (here is an [example usage](https://github.com/slackapi/python-slack-sdk/releases/tag/v3.19.0) from one of the Slack devs if you are interested)

@dorschw Perhaps I should add a comment next to each field to explain the meaning?

File: Packs/Slack/Integrations/SlackV3/SlackV3_test.py [hunk 27]
```diff
@@ -3291,26 +3289,53 @@ def test_send_message_to_destinations(mocker):
     assert args['body']['text']
 
 
-def test_send_file_to_destinations(mocker):
+def test_send_file_to_destinations_request_args(mocker):
+    """
+    Given:
+        - Sending file to a channel
```
Comment (dorschw) [inline Packs/Slack/Integrations/SlackV3/SlackV3_test.py:3295]:
```suggestion
        - A file to send
```

File: Packs/Slack/Integrations/SlackV3/SlackV3_test.py [hunk 28]
```diff
@@ -3291,26 +3289,53 @@ def test_send_message_to_destinations(mocker):
...
-    assert args['body']['filename'] == 'name'
+    assert 'http_verb' not in request_args
+    assert request_args['file_upload_params']['file'] == 'yo'
+    assert request_args['file_upload_params']['filename'] == 'name'
+
+
+def test_send_file_to_destinations_sdk_client_args(mocker):
+    """
+    Given:
+        - Sending file to a specifc thread in a channel
```
Comment (dorschw) [inline Packs/Slack/Integrations/SlackV3/SlackV3_test.py:3319]:
```suggestion
        - A file to send
```

File: Packs/Slack/Integrations/SlackV3/SlackV3_test.py [hunk 29]
```diff
@@ -3291,26 +3289,53 @@ def test_send_message_to_destinations(mocker):
     assert args['body']['text']
 
 
-def test_send_file_to_destinations(mocker):
+def test_send_file_to_destinations_request_args(mocker):
+    """
```
Comment (dorschw) [inline Packs/Slack/Integrations/SlackV3/SlackV3_test.py:3293]:
```suggestion
    """
    This one mocks ..., while <other one> mocks ... 
```

File: Packs/Slack/Integrations/SlackV3/SlackV3_test.py (hunk 24 above)
Comment (kamalq97) [reply Packs/Slack/Integrations/SlackV3/SlackV3_test.py:3534]: It was `demisto.result` before, now it is `CommandResults` so I had to update the test to access the `HumanReadable` attribute of the new object

File: Packs/Slack/Integrations/SlackV3/SlackV3_test.py (hunk 25 above)
Comment (kamalq97) [reply Packs/Slack/Integrations/SlackV3/SlackV3_test.py:5274]: I only changed it since we no longer have deprecated methods in our SlackV3 integration.

Other than that, there is no functional reason, so I will put back the original error message if you would like.

File: Packs/Slack/Integrations/SlackV3/SlackV3.py (hunk 21 above)
Comment (kamalq97) [reply Packs/Slack/Integrations/SlackV3/SlackV3.py]: Fixed in commit 7c5b1ee65136fc655e92a3456301506de163b9e6

File: Packs/Slack/Integrations/SlackV3/SlackV3.py (hunk 22 above)
Comment (kamalq97) [reply Packs/Slack/Integrations/SlackV3/SlackV3.py]: Fixed in commit 7c5b1ee65136fc655e92a3456301506de163b9e6

File: Packs/Slack/Integrations/SlackV3/SlackV3_test.py (hunk 27 above)
Comment (kamalq97) [reply Packs/Slack/Integrations/SlackV3/SlackV3_test.py]: Improved doc string in commit 7c5b1ee65136fc655e92a3456301506de163b9e6

File: Packs/Slack/Integrations/SlackV3/SlackV3_test.py (hunk 28 above)
Comment (kamalq97) [reply Packs/Slack/Integrations/SlackV3/SlackV3_test.py]: Improved doc string in commit 7c5b1ee65136fc655e92a3456301506de163b9e6

File: Packs/Slack/Integrations/SlackV3/SlackV3_test.py (hunk 29 above)
Comment (kamalq97) [reply Packs/Slack/Integrations/SlackV3/SlackV3_test.py:3293]: Added a sentence to the beginning of the doc string in commit 7c5b1ee65136fc655e92a3456301506de163b9e6 to differentiate between the two tests

File: Packs/Slack/Integrations/SlackV3/SlackV3_test.py [hunk 30]
```diff
@@ -5232,29 +5257,23 @@ def test_get_war_room_url_with_xsiam_from_alert_war_room(self, mocker):
         assert get_war_room_url(url) == expected_war_room_url
 
 
-def test_send_file_deprecated_endpoint(mocker, requests_mock):
+def test_send_file_api_exception(mocker):
     """
     Given:
-        - Slack client of newaly created app (May 8,2024).
+        - Slack API non-okay response.
```
[the remark this answers is not in the document]
Comment (kamalq97) [reply Packs/Slack/Integrations/SlackV3/SlackV3_test.py]: Improved doc string in commit 7c5b1ee65136fc655e92a3456301506de163b9e6

File: Packs/Slack/Integrations/SlackV3/SlackV3.py (hunk 19 above)
Comment (kamalq97) [reply Packs/Slack/Integrations/SlackV3/SlackV3.py]: Improved comments and added link to Slack SDK docs in commit 7c5b1ee65136fc655e92a3456301506de163b9e6

File: Packs/Slack/Integrations/SlackV3/SlackV3.py (hunk 18 above)
Comment (kamalq97) [reply Packs/Slack/Integrations/SlackV3/SlackV3.py]: Improved doc string in commit 7c5b1ee65136fc655e92a3456301506de163b9e6

File: Packs/Slack/Integrations/SlackV3/SlackV3.py (hunk 20 above)
Comment (kamalq97) [reply Packs/Slack/Integrations/SlackV3/SlackV3.py]: I have now added the file name in the error message in commit 86a9eb17e04ab27552e682f68301ea4abf685423.
Note that the explanation of the error is concatenated with the error message under the `except ...` block a few lines below

File: Packs/Slack/pack_metadata.json (hunk 26 above)
Comment (kamalq97) [reply Packs/Slack/pack_metadata.json]: I have bumped up the minor version from `3.4.19` to `3.5.0` in commit 86a9eb17e04ab27552e682f68301ea4abf685423 using the demisto-sdk

File: Packs/Slack/Integrations/SlackV3/SlackV3.py (hunk 14 above)
Comment (dorschw) [reply Packs/Slack/Integrations/SlackV3/SlackV3.py:94]: a comment will be great, yes

Comment (kamalq97) [reply Packs/Slack/Integrations/SlackV3/SlackV3.py]: Added comments in commit 2d3ae24d9b52ec8d660e15df52420ed82c545827

File: Packs/Slack/Integrations/SlackV3/SlackV3.py (hunk 13 above)
Comment (kamalq97) [reply Packs/Slack/Integrations/SlackV3/SlackV3.py]: Moved it just above the function where it is used in commit 2d3ae24d9b52ec8d660e15df52420ed82c545827

Perhaps we should remove this dictionary all together and just display the error "as is" from the response?

Comment (kamalq97) [reply Packs/Slack/Integrations/SlackV3/SlackV3.py]: I have resolved this conversation based on our discussion in Slack

**Review by dorschw — APPROVED** [review]

**Review by shmuel44 — APPROVED** [review]
